internal/pod/workerpodmanager.go

```diff
@@ -36,16 +36,15 @@ type WorkerPodManager interface {
 	ListWorkerPodsOnNode(ctx context.Context, nodeName string) ([]v1.Pod, error)
 	LoaderPodTemplate(ctx context.Context, nmc client.Object, nms *kmmv1beta1.NodeModuleSpec) (*v1.Pod, error)
 	UnloaderPodTemplate(ctx context.Context, nmc client.Object, nms *kmmv1beta1.NodeModuleStatus) (*v1.Pod, error)
+	IsLoaderPod(p v1.Pod) bool
```

Comment:
I think all those function can receive a Pod pointer, instead of a Pod structure. Passing it as a struct mean copying it, and it is a large structure. There are a lot of functions , even in kuberenetes packages, that receive parameters as pointers, event though they are not changing them

Reply:
Fixed.